sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c

```diff
@@ -59,6 +59,21 @@ void _receive(ng_pktsnip_t *pkt)
     uint8_t *dispatch;
     ng_netreg_entry_t *entry;
 
+    /* seize payload as a temporary variable */
+    payload = ng_pktbuf_start_write(pkt);   /* need to duplicate since pkt->next
```

Comment:
So mainly you need this outside the `dispatch[0] == NG_SIXLOWPAN_UNCOMPRESSED` dispatch, right? The rest seems really redundant to me, comparing lines  79-96. I just had a quick look, please correct me if I'm wrong!